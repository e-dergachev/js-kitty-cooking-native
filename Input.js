import React from 'react';
import { StyleSheet, TextInput, View } from 'react-native';

function Input(props) {

  const styles = StyleSheet.create({
    input: {
      backgroundColor: props.scheme.color2,
      borderWidth: 1,
      alignItems: 'center',
      width: '85%',
      height: '25%',
      marginTop: 15,
      marginBottom: 15,
      borderColor: props.scheme.color5,
    },
    search: {
      width: '95%',
      borderBottomWidth: 2,
      borderColor: props.scheme.color5,
      color: props.scheme.color6,
    },
  });

  return (
    <View style={styles.input}>
      <TextInput
        style={styles.search} 
        placeholder="What dish are you going to cook?"
        placeholderTextColor="#808080"
      />
    </View>
  );
}

export default Input;
