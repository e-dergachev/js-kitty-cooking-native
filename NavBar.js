import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import Constants from 'expo-constants';

function NavBar(props) {

    const styles = StyleSheet.create({
        foldedNavBar: {
            backgroundColor: props.scheme.color3,
            borderWidth: 1,
            alignItems: 'center',
            justifyContent: 'center',
            height: '9%',
            marginTop: Constants.statusBarHeight + 2, //marginTop: 27
            width: '99%',
            borderColor: props.scheme.color5,
        },
        msg: {
            color: props.scheme.color7,
            fontSize: 15,
        }
    });

    return (
        <View style={styles.foldedNavBar}>
            <Text style={styles.msg}>Click to select cuisines</Text>
        </View>
    );
}

export default NavBar;
