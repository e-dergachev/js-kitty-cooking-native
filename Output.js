import React, { useState } from 'react';
import { StyleSheet, View, TouchableHighlight, Image, Text } from 'react-native';
import { FontAwesome } from '@expo/vector-icons';

function Output(props) {

    const [pressStatus, setPressStatus] = useState({random: false, lavender: false, green: false, pink: false});

    const styles = StyleSheet.create({
        output: {
            backgroundColor: props.scheme.color2,
            borderWidth: 1,
            alignItems: 'center',
            width: '90%',
            borderColor: props.scheme.color5,
            marginBottom: 15,
            flex: 1,
        },
        outputCap: {
            flexDirection: 'row-reverse',
            width: '100%',
        },
        button: {
            borderColor: props.scheme.color5,
            borderRadius: 7,
            marginTop: 4,
            marginRight: 4,
            height: 30,
            width: 80,
            justifyContent: 'center',
            alignItems: 'center',
            backgroundColor: props.scheme.color4,          
        },
        buttonNotPressed: {
            borderWidth: 1,
        },        
        buttonPressed: {
            borderWidth: 2,
        },
        buttonText: {
            color: props.scheme.color8,
        },
        appTitle: {
            fontSize: 30,
            color: props.scheme.color6,
            fontFamily: 'indie-flower',
            marginBottom: 3,
        },
        simpleText: {
            color: props.scheme.color6,
            textAlign: 'center',
        },
        kittyBox: {
            flex: 1,
            justifyContent: 'center',
            marginBottom: 5,
        },
        kitty: {
            height: '70%'//110,
        },
        bottom: {
            justifyContent: "flex-end",
        },
        themes: {
            flexDirection: 'row',
            justifyContent: 'center',
            marginTop: 5,
        },
        theme: {
            borderWidth: 1,
            borderColor: props.scheme.color5,
            width: 65, 
            height: 45,
        },
        themePressed: {
            borderWidth: 2,
            borderColor: props.scheme.color5,
            width: 65, 
            height: 45,
        },
        lavender: {
            backgroundColor: '#e6e6fa',
        },
        green: {
            backgroundColor: '#b3ff99',
            marginRight: 8,
            marginLeft: 8,
        },
        pink: {
            backgroundColor: '#efa7c0',
        },
        legal: {
            fontSize: 11,
            textAlign: 'center',
            color: props.scheme.color6,
            marginBottom: 5,
            marginTop: 5,
        },
    });

    return (
        <View style={styles.output}>
            <View style={styles.outputCap}>
                <TouchableHighlight
                    onPress={() => {}}
                    onHideUnderlay={() => setPressStatus(prevState => {return {...prevState, random: false}})}
                    onShowUnderlay={() => setPressStatus(prevState => {return {...prevState, random: true}})}
                    style={{...pressStatus['random'] ? styles.buttonPressed : styles.buttonNotPressed, ...styles.button}}
                    underlayColor={props.scheme.color4}              
                >
                    <View>
                        <Text style={styles.buttonText}>Random <FontAwesome name="random" color={props.scheme.color5} /></Text>
                    </View>
                </TouchableHighlight>
            </View>
            {props.fontLoaded ? (<Text style={styles.appTitle}>JS Kitty Cooking</Text>) : null}
            <Text style={styles.simpleText}>Old recipes from ancient cookbooks</Text>
            <View style={styles.kittyBox}>
                <Image
                    source={require('./assets/kitty.png')}
                    style={styles.kitty}
                    resizeMode='contain'
                />
            </View>
            <View style={styles.bottom}>
                <Text style={styles.simpleText}>Choose a color scheme:</Text>
                <View style={styles.themes}>
                    <TouchableHighlight 
                        onPress={() => props.setColorScheme("lavender")}
                        onHideUnderlay={() => setPressStatus(prevState => {return {...prevState, lavender: false}})}
                        onShowUnderlay={() => setPressStatus(prevState => {return {...prevState, lavender: true}})}
                        style={{...pressStatus['lavender'] ? styles.themePressed : styles.theme, ...styles.lavender}}
                        underlayColor='#e6e6fa'
                    >
                        <View />
                    </TouchableHighlight>
                    <TouchableHighlight 
                        onPress={() => props.setColorScheme("green")}
                        onHideUnderlay={() => setPressStatus(prevState => {return {...prevState, green: false}})}
                        onShowUnderlay={() => setPressStatus(prevState => {return {...prevState, green: true}})}
                        style={{...pressStatus['green'] ? styles.themePressed : styles.theme, ...styles.green}}
                        underlayColor='#b3ff99'
                    >
                        <View />
                    </TouchableHighlight>
                    <TouchableHighlight 
                        onPress={() => props.setColorScheme("pink")}
                        onHideUnderlay={() => setPressStatus(prevState => {return {...prevState, pink: false}})}
                        onShowUnderlay={() => setPressStatus(prevState => {return {...prevState, pink: true}})}
                        style={{...pressStatus['pink'] ? styles.themePressed : styles.theme, ...styles.pink}}
                        underlayColor='#efa7c0'
                    >
                        <View />
                    </TouchableHighlight>
                </View>
                <Text style={styles.legal}>All the recipes are taken from the public domain books on Guttenberg.com</Text>
            </View>
        </View>
    );
}

export default Output;
