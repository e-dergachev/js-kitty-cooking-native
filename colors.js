const colors = {   //colorSchemes
    lavender: {
        color1: '#e6e6fa', //hsl(240, 67%, 94%) main background
        color2: '#edeeff', //hsl(237, 100%, 96%) input/output, even suggestions backgrounds
        color3: '#aeafc2', //hsl(237, 14%, 72%) folded navbar, selected cuisines backgrounds
        color4: '#ddddf8', //hsl(240, 66%, 92%) buttons, unfolded navbar, unselected cuisines, odd suggestions backgrounds
        color5: '#5d4300', //hsl(43, 100%, 18%) all borders, font-awesome
        color6: '#291e00', //hsl(44, 100%, 8%) main font
        color7: '#332500', //hsl(44, 100%, 10%) folded/unfolded navbar & cuisines fonts
        color8: '#3d2d00' //hsl(44, 100%, 12%) output-cap font
    },
    green: {
        color1: '#b3ff99', //hsl(105, 100%, 80%)
        color2: '#dff8d5', //hsl(103, 71%, 90%)
        color3: '#9abe8c', //hsl(103, 28%, 65%)
        color4: '#a4fe85', //hsl(105, 99%, 76%)
        color5: '#5d4300', //hsl(43, 100%, 18%) the same for all schemes
        color6: '#291e00', //hsl(44, 100%, 8%) the same for all schemes
        color7: '#332500', //hsl(44, 100%, 10%) the same for all schemes
        color8: '#3d2d00' //hsl(44, 100%, 12%) the same for all schemes
    },
    pink: {
        color1: '#efa7c0', //hsl(339, 69%, 80%)
        color2: '#ffe3f4', //hsl(324, 100%, 95%)
        color3: '#b6728b', //hsl(338, 32%, 58%)
        color4: '#eb98b5', //hsl(339, 68%, 76%)
        color5: '#5d4300', //hsl(43, 100%, 18%) the same for all schemes
        color6: '#291e00', //hsl(44, 100%, 8%) the same for all schemes
        color7: '#332500', //hsl(44, 100%, 10%) the same for all schemes
        color8: '#3d2d00' //hsl(44, 100%, 12%) the same for all schemes
    }
};

module.exports = colors;
