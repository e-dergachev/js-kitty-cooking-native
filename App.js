import React, { useState, useEffect } from 'react';
import { StyleSheet, View } from 'react-native';
import * as Font from 'expo-font';
import Input from './Input';
import Output from './Output';
import NavBar from './NavBar';
import colors from './colors.js';
import { Asset } from 'expo-asset';
import * as FileSystem from 'expo-file-system';

function App() {
  const [colorScheme, setColorScheme] = useState("lavender");
  const [fontLoaded, setFontLoadingStatus] = useState(false);
  const [navFolded, setNavFolded] = useState(true);
  const [cuisines, setCuisines] = useState({General: true, Vegetarian: true, American: true, French: true, Indian: true, Italian: true, Jewish: true});
  const [dish, setDish] = useState({});

  async function loadFont() {
    await Font.loadAsync({
      'indie-flower': require('./assets/fonts/IndieFlower-Regular.ttf'),
    });
    setFontLoadingStatus(true);
  }

  function loadDB() {
    FileSystem.downloadAsync(
      Asset.fromModule(require('./assets/db/recipes.sqlite3')).uri, `${FileSystem.documentDirectory}SQLite/recipes.sqlite3`)
      .then(({ uri }) => {
        console.log('Finished downloading to ', uri)
      })
      .catch(error => error);
  }

  useEffect(() => {
    loadFont();
    loadDB();
  }, []); //the empty array tells it to never re-render

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      backgroundColor: colors[colorScheme].color1,
    },
  });

  return (
    <View style={styles.container}>
      <NavBar 
        scheme={colors[colorScheme]}
        navFolded={navFolded}
        setNavFolded={setNavFolded}
        cuisines={cuisines}
        setCuisines={setCuisines}
      />
      <Input 
        scheme={colors[colorScheme]}
        navFolded={navFolded}
        cuisines={cuisines}
        setDish={setDish}
      />
      <Output 
        scheme={colors[colorScheme]} //leave it here instead of using colorScheme directly to keep the code more clear
        colorScheme={colorScheme}
        setColorScheme={setColorScheme}
        fontLoaded={fontLoaded}
        cuisines={cuisines}
        dish={dish}
        setDish={setDish}
      />
    </View>
  );
}

export default App;
