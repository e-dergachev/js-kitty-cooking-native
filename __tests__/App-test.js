import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import colors from '../colors';
import NavBar from '../NavBar';
import Input from '../Input';
import Output from '../Output';

const cuisines = {General: true, Vegetarian: true, American: true, French: true, Indian: true, Italian: true, Jewish: true};

const dish = {
  name: "Potato Pie",
  recipe: "Slice potatoes and onions, stew with a little water until nearly done, put into a pie-dish, flavour with herbs, pepper, and salt, add a little soaked tapioca and very little butter, cover with short wheatmeal crust, and bake 1 hour. To make a very plain pie-crust use about 2 oz. of butter or a proportionate quantity of Allinson frying oil to 1 lb. of wheatmeal. Roll or touch with the fingers as little as possible, and mix with milk instead of water. Eat this pie with green vegetables.",
  source: "The Allinson Vegetarian Cookery Book, by Thomas R. Allinson, 1915",
  cuisine: "Vegetarian"
};

describe('Snapshot Unit Tests', () => {

  it('renders app wrapper correctly', () => { //it gives that annoying 'wrap stuff in act(...)' warning but passes
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders navbar correctly', () => {
    const tree = renderer.create(<NavBar 
      scheme = {colors['lavender']}
      cuisines = {cuisines}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders input correctly', () => {
    const tree = renderer.create(<Input 
      scheme = {colors['lavender']}
      cuisines = {cuisines}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('renders output correctly', () => {
    const tree = renderer.create(<Output 
      scheme = {colors['lavender']}
      cuisines = {cuisines}
      dish = {dish}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });  

});
