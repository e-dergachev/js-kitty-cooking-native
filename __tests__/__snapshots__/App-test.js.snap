// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Unit Tests renders app wrapper correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e6e6fa",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#aeafc2",
        "borderColor": "#5d4300",
        "borderWidth": 1,
        "height": 55,
        "justifyContent": "center",
        "marginTop": 22,
        "width": "99%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#332500",
          "fontSize": 15,
        }
      }
    >
      Press to select cuisines
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 15,
        "marginTop": 15,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#edeeff",
          "borderColor": "#5d4300",
          "borderWidth": 1,
          "flex": 1,
          "height": 152,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="What dish are you going to cook?"
        placeholderTextColor="#808080"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomWidth": 2,
            "borderColor": "#5d4300",
            "color": "#291e00",
            "height": 30,
            "width": "95%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <RCTScrollView
        persistentScrollbar={true}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View />
      </RCTScrollView>
    </View>
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ddddf8",
            "borderColor": "#5d4300",
            "borderRadius": 7,
            "borderWidth": 1,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 5,
            "width": 40,
          }
        }
      >
        <Text />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ddddf8",
            "borderColor": "#5d4300",
            "borderRadius": 7,
            "borderWidth": 1,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 5,
            "width": 40,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#edeeff",
        "borderColor": "#5d4300",
        "borderWidth": 1,
        "flex": 1,
        "marginBottom": 15,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row-reverse",
          "paddingBottom": 5,
          "paddingRight": 5,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ddddf8",
            "borderColor": "#5d4300",
            "borderRadius": 7,
            "borderWidth": 1,
            "height": 30,
            "justifyContent": "center",
            "marginRight": 5,
            "marginTop": 5,
            "width": 80,
          }
        }
      >
        <View
          style={null}
        >
          <Text
            style={
              Object {
                "color": "#3d2d00",
              }
            }
          >
            Random 
            <Text />
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.6,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#291e00",
            "textAlign": "center",
          }
        }
      >
        Old recipes from ancient cookbooks
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": "70%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 0.8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#291e00",
            "textAlign": "center",
          }
        }
      >
        Choose a color scheme:
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 5,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#e6e6fa",
              "borderColor": "#5d4300",
              "borderWidth": 1,
              "height": 45,
              "width": 65,
            }
          }
        >
          <View
            style={null}
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#b3ff99",
              "borderColor": "#5d4300",
              "borderWidth": 1,
              "height": 45,
              "marginLeft": 8,
              "marginRight": 8,
              "width": 65,
            }
          }
        >
          <View
            style={null}
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#efa7c0",
              "borderColor": "#5d4300",
              "borderWidth": 1,
              "height": 45,
              "width": 65,
            }
          }
        >
          <View
            style={null}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#291e00",
            "fontSize": 11,
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        All the recipes are taken from the public domain books on Guttenberg.com
      </Text>
    </View>
  </View>
</View>
`;

exports[`Snapshot Unit Tests renders input correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 15,
      "marginTop": 92,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#edeeff",
        "borderColor": "#5d4300",
        "borderWidth": 1,
        "flex": 1,
        "height": 152,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="What dish are you going to cook?"
      placeholderTextColor="#808080"
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomWidth": 2,
          "borderColor": "#5d4300",
          "color": "#291e00",
          "height": 30,
          "width": "95%",
        }
      }
      underlineColorAndroid="transparent"
    />
    <RCTScrollView
      persistentScrollbar={true}
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ddddf8",
          "borderColor": "#5d4300",
          "borderRadius": 7,
          "borderWidth": 1,
          "height": 30,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginLeft": 5,
          "width": 40,
        }
      }
    >
      <Text />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ddddf8",
          "borderColor": "#5d4300",
          "borderRadius": 7,
          "borderWidth": 1,
          "height": 30,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginLeft": 5,
          "width": 40,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;

exports[`Snapshot Unit Tests renders navbar correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ddddf8",
      "borderColor": "#5d4300",
      "borderWidth": 1,
      "height": 420,
      "marginTop": 22,
      "position": "absolute",
      "width": "99%",
      "zIndex": 100,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#332500",
          "fontSize": 15,
          "marginTop": 5,
        }
      }
    >
      Press cuisines to select/unselect them
    </Text>
    <Text
      style={
        Object {
          "color": "#332500",
          "fontSize": 15,
          "marginBottom": 5,
        }
      }
    >
      Press the bar again to close it
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#aeafc2",
          "borderColor": "#5d4300",
          "borderRadius": 7,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 5,
          "width": 225,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#332500",
            "fontSize": 15,
          }
        }
      >
        General
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#aeafc2",
          "borderColor": "#5d4300",
          "borderRadius": 7,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 5,
          "width": 225,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#332500",
            "fontSize": 15,
          }
        }
      >
        Vegetarian
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#aeafc2",
          "borderColor": "#5d4300",
          "borderRadius": 7,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 5,
          "width": 225,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#332500",
            "fontSize": 15,
          }
        }
      >
        American
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#aeafc2",
          "borderColor": "#5d4300",
          "borderRadius": 7,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 5,
          "width": 225,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#332500",
            "fontSize": 15,
          }
        }
      >
        French
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#aeafc2",
          "borderColor": "#5d4300",
          "borderRadius": 7,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 5,
          "width": 225,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#332500",
            "fontSize": 15,
          }
        }
      >
        Indian
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#aeafc2",
          "borderColor": "#5d4300",
          "borderRadius": 7,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 5,
          "width": 225,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#332500",
            "fontSize": 15,
          }
        }
      >
        Italian
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#aeafc2",
          "borderColor": "#5d4300",
          "borderRadius": 7,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 5,
          "width": 225,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#332500",
            "fontSize": 15,
          }
        }
      >
        Jewish
      </Text>
    </View>
  </View>
</View>
`;

exports[`Snapshot Unit Tests renders output correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#edeeff",
      "borderColor": "#5d4300",
      "borderWidth": 1,
      "flex": 1,
      "marginBottom": 15,
      "width": "90%",
    }
  }
>
  <RCTScrollView
    persistentScrollbar={true}
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row-reverse",
            "paddingBottom": 5,
            "paddingRight": 5,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ddddf8",
              "borderColor": "#5d4300",
              "borderRadius": 7,
              "borderWidth": 1,
              "height": 30,
              "justifyContent": "center",
              "marginRight": 5,
              "marginTop": 5,
              "width": 80,
            }
          }
        >
          <View
            style={null}
          >
            <Text
              style={
                Object {
                  "color": "#3d2d00",
                }
              }
            >
              Random 
              <Text />
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ddddf8",
              "borderColor": "#5d4300",
              "borderRadius": 7,
              "borderWidth": 1,
              "height": 30,
              "justifyContent": "center",
              "marginRight": 5,
              "marginTop": 5,
              "width": 58,
            }
          }
        >
          <View
            style={null}
          >
            <Text
              style={
                Object {
                  "color": "#3d2d00",
                }
              }
            >
              Clear 
              <Text />
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 2,
              "borderColor": "#5d4300",
              "height": 30,
              "justifyContent": "center",
              "marginLeft": "auto",
              "marginTop": 5,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#3d2d00",
              }
            }
          >
            Cuisine: 
            Vegetarian
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "marginBottom": 10,
            "marginTop": 9,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#291e00",
              "fontSize": 16,
            }
          }
        >
          Potato Pie
        </Text>
        <Text
          style={
            Object {
              "color": "#291e00",
            }
          }
        >
          -------------------------------------
        </Text>
        <Text
          style={
            Object {
              "color": "#291e00",
            }
          }
        >
          Slice potatoes and onions, stew with a little water until nearly done, put into a pie-dish, flavour with herbs, pepper, and salt, add a little soaked tapioca and very little butter, cover with short wheatmeal crust, and bake 1 hour. To make a very plain pie-crust use about 2 oz. of butter or a proportionate quantity of Allinson frying oil to 1 lb. of wheatmeal. Roll or touch with the fingers as little as possible, and mix with milk instead of water. Eat this pie with green vegetables.
        </Text>
        <Text
          style={
            Object {
              "color": "#291e00",
            }
          }
        >
          -------------------------------------
        </Text>
        <Text
          style={
            Object {
              "color": "#291e00",
            }
          }
        >
          The Allinson Vegetarian Cookery Book, by Thomas R. Allinson, 1915
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
